mim/commands/install.py

```diff
@@ -1,580 +1,303 @@
-        try:
-            install_from_wheel(target_pkg, target_version, find_url, timeout,
-                               is_user_dir)
-        except RuntimeError as error:
-            if target_pkg in PKG2PROJECT:
-                find_url = f'{DEFAULT_URL}/{PKG2PROJECT[target_pkg]}.git'
-                if target_version:
-                    target_pkg = f'{target_pkg}=={target_version}'
-                if is_yes:
-                    install(target_pkg, find_url, timeout, is_yes, is_user_dir)
-                else:
-                    confirm_msg = (f'install {target_pkg} from wheel, but it '
-                                   'failed. Do you want to build it from '
-                                   'source if possible?')
-                    if click.confirm(confirm_msg):
-                        install(target_pkg, find_url, timeout, is_yes,
-                                is_user_dir)
-                    else:
-                        raise RuntimeError(
-                            highlighted_error(
-                                f'Failed to install {target_pkg}.'))
-            else:
-                raise RuntimeError(highlighted_error(error))
-
-    echo_success(f'Successfully installed {target_pkg}.')
-
-
-def looks_like_path(name: str) -> bool:
-    """Checks whether the string "looks like" a path on the filesystem.
-
-    This does not check whether the target actually exists, only judge from the
-    appearance.
+def install(
+    install_args: List[str],
+    index_url: Optional[str] = None,
+    is_yes: bool = False,
+) -> Any:
+    """Install the package via pip and add 'mim' extra requirements for
+    OpenMMLab package during pip install process.
 
     Args:
-        name (str): The string to be checked.
+        install_args (list): List of arguments passed to `pip install`.
+        index_url (str, optional): The pypi index url.
+        is_yes (bool, optional): Deprecated, will have no effect. Reserved for
+            interface compatibility only.
     """
-    if osp.sep in name:
-        return True
-    if osp.altsep is not None and osp.altsep in name:
-        return True
-    if name.startswith('.'):
-        return True
-    return False
+    # add mmcv-full find links by default
+    install_args += ['-f', get_mmcv_full_find_link()]
+
+    if index_url is not None and '-i' not in install_args:
+        install_args += ['-i', index_url]
+
+    patch_mm_distribution: Callable = patch_pkg_resources_distribution
+    try:
+        # pip>=22.1 have two distribution backend: pkg_resources and importlib.
+        from pip._internal.metadata import _should_use_importlib_metadata  # type: ignore # isort: skip # noqa: E501
+        if _should_use_importlib_metadata():
+            patch_mm_distribution = patch_importlib_distribution
+    except ImportError:
+        pass
 
+    with patch_mm_distribution(index_url):
+        # We can use `create_command` method since pip>=19.3 (2019/10/14).
```

Comment:
Should we also add the restriction for pip in requirements/install.txt?

Reply:
Done!